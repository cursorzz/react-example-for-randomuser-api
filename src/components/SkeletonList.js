import React from 'react'
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Skeleton from '@material-ui/lab/Skeleton';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    '& div': {
      marginRight: 10

    }
  },
});

export function SkeletonList({ rows }) {
  const classes = useStyles();

  return (rows.map((row, index) => {
    return (
      <TableRow key={index}>
        <TableCell component="th" scope="row" className={classes.root}>
          <Skeleton variant="circle" width={40} height={40} />
          <Skeleton variant="text" width={"50%"} height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
        <TableCell component="th" scope="row">
          <Skeleton variant="text" height={40} />
        </TableCell>
      </TableRow>)
  })
  )
}