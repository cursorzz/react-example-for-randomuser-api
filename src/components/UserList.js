import React from 'react'
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Avatar from '@material-ui/core/Avatar';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  root: {
    display: "flex",
    alignItems: "center",
    '& div': {
      marginRight: 10

    }
  },
});

export function UserList({ rows }) {
  const classes = useStyles();

  const getName = (result) => {
    const { name } = result
    return `${name.title} ${name.first} ${name.last}`
  }

  return (rows.map(row => {
    return (<TableRow key={row.login.uuid}>
      <TableCell component="th" scope="row" className={classes.root}>
        <Avatar alt={getName(row)} src={row.picture.thumbnail} />
        {getName(row)}
      </TableCell>
      <TableCell align="left" component="th">{row.location.city}</TableCell>
      <TableCell align="left" component="th">{row.location.state}</TableCell>
      <TableCell align="left" component="th">{row.email}</TableCell>
    </TableRow>)
  }))
}