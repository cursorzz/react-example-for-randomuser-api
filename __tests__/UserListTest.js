import React from "react";
import { UserList } from "../src/components/UserList";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const users = [
    {
      gender: "female",
      name: { title: "Mrs", first: "Nurdan", last: "Karab√∂cek" },
      location: {
        street: { number: 9334, name: "Doktorlar Cd" },
        city: "Rize",
        state: "Hatay",
        country: "Turkey",
        postcode: 41411,
        coordinates: { latitude: "76.1898", longitude: "93.4928" },
        timezone: { offset: "+6:00", description: "Almaty, Dhaka, Colombo" }
      },
      email: "nurdan.karabocek@example.com",
      login: {
        uuid: "b77ce2a6-5521-4627-a077-ce8b1367a0b5",
        username: "lazyrabbit925",
        password: "joel",
        salt: "vheAlDPb",
        md5: "60120d395e7f1c502e6d6880b3e999e4",
        sha1: "bde493c5cf9ce145c61643a4742615846ecc3687",
        sha256:
          "006214ba5d1cd8f9781fa9d1e95de6083cefc1532492af620053cb0a33520c85"
      },
      dob: { date: "1973-05-12T08:05:47.336Z", age: 47 },
      registered: { date: "2014-12-05T11:02:40.986Z", age: 6 },
      phone: "(134)-607-3573",
      cell: "(737)-003-0643",
      id: { name: "", value: null },
      picture: {
        large: "https://randomuser.me/api/portraits/women/9.jpg",
        medium: "https://randomuser.me/api/portraits/med/women/9.jpg",
        thumbnail: "https://randomuser.me/api/portraits/thumb/women/9.jpg"
      },
      nat: "TR"
    }
  ];
  const tree = renderer.create(<UserList rows={users} />).toJSON();
  expect(tree).toMatchSnapshot();
});
