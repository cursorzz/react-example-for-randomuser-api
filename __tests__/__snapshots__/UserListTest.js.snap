// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr
  className="MuiTableRow-root"
>
  <th
    aria-sort={null}
    className="MuiTableCell-root makeStyles-root-1"
    scope="row"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle"
    >
      <img
        alt="Mrs Nurdan Karaböcek"
        className="MuiAvatar-img"
        src="https://randomuser.me/api/portraits/thumb/women/9.jpg"
      />
    </div>
    Mrs Nurdan Karaböcek
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    Rize
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    Hatay
  </th>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignLeft"
  >
    nurdan.karabocek@example.com
  </th>
</tr>
`;
